### PR TITLE
Made adjustment to make scopes operate as expected.

### PR DESCRIPTION
**Pull request by rigel-eva**
Also updated the site, authorize url, and token url parameters to futureproof the gem going forward.

This fixes the issue I posted about earlier, [Issue 13 - Adding New scopes results in error](https://github.com/WebTheoryLLC/omniauth-twitch/issues/13)

### CODE REVIEW

**Comment by jcgertig** [comment]
I cant look at this today so the earliest will be the morning.

**Comment by rigel-eva** [comment]
Fair enough, thank you for your prompt response!

**Comment by jcgertig** [comment]
@rigel-eva does this still do what you want?

File: lib/omniauth/strategies/twitch.rb [hunk 1]
```diff
@@ -20,6 +20,12 @@ class Twitch < OmniAuth::Strategies::OAuth2
 
       option :authorize_options, [:scope]
 
+      def request_phase
+        redirect client.auth_code.
+          authorize_url({ redirect_uri: callback_url }.merge(authorize_params)).
```
Comment (jcgertig) [inline lib/omniauth/strategies/twitch.rb:25]: should `authorize_params` be `authorize_options`??

**Review by jcgertig — CHANGES_REQUESTED** [review]

File: lib/omniauth/strategies/twitch.rb (hunk 1 above)
Comment (rigel-eva) [reply lib/omniauth/strategies/twitch.rb:25]: No. `authorize_options` is an `Array` and it's expecting an `OmniAuth::Strategy::Options` which `authorize_params` generates.

Comment (jcgertig) [reply lib/omniauth/strategies/twitch.rb:25]: okay but does the refactor still do what you expect?

Comment (rigel-eva) [reply lib/omniauth/strategies/twitch.rb:25]: Yes it does. It allows for the use of additional scopes, as in here: https://github.com/rigel-eva/CmdrShepard/blob/5fa4b5ad2aa31a31a80b4d2c7df2b36d89f4fc56/webserver/config/initializers/omniauth.rb#L7

**Comment by krtschmr** [comment]
when is this getting merged?

**Pull request merged**